test/Tests/src/Data/Vector_Spec.enso

```diff
@@ -49,7 +49,7 @@ spec = Test.group "Vectors" <|
         built_from_js = Vector.from_polyglot_array generate_js_array
         built_from_js . should_equal [1, 2, 3, 4, 5]
         built_from_py = Vector.from_polyglot_array generate_py_array
-        built_from_py . should_equal [1, 2, 3, 4, 5]
+        built_from_py . should_equal [1, 2, 3, 4, Nothing]
```

Comment:
Yup, I believe this is desirable. Foreign `null`s are equal to Enso `Nothing`. Btw. this is not what Graal.js does. If you:
```js
var value = foreignCall();
if (value === null) print('===');
if (value == null) print("==");
```
then only the second condition may be `true`! In Graal.js foreign `null`s are not `===` with JavaScript `null`, but are only `==` with it. I've been burned by this _feature_ so many times - I suggest in Enso `Nothing` is equal with any foreign _Nothing_/`null`.